pkg/cmd/pulumi/package_publish.go

```diff
@@ -0,0 +1,176 @@
+// Copyright 2016-2022, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/env"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/executable"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/logging"
+	"github.com/spf13/cobra"
+
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/cmdutil"
+)
+
+func newPackagePublishCmd() *cobra.Command {
+	var publCmd publishCmd
+	cmd := &cobra.Command{
+		Use:    "publish-sdk <package> <language>",
+		Args:   cobra.ExactArgs(2),
+		Short:  "Publish a package SDK to supported package registries.",
+		Hidden: !env.Dev.Value(),
+		Run: cmdutil.RunFunc(func(cmd *cobra.Command, args []string) error {
+			ctx := commandContext()
+			return publCmd.Run(ctx, args)
+		}),
+	}
+	cmd.PersistentFlags().StringVar(&publCmd.Path, "path", "",
+		`The path to the root of your package.
+	Example: ./sdk/nodejs
+	`)
+	return cmd
+}
+
+type publishCmd struct {
+	Path string
+}
+
+func (cmd *publishCmd) Run(ctx context.Context, args []string) error {
+	pkg := args[0]
+	lang := args[1]
+
+	logging.V(1).Infof("publishing package %s to the %s registry \n", pkg, lang)
+	switch lang {
+	case "nodejs":
+		err := publishToNPM(pkg, cmd.Path)
+		if err != nil {
+			return err
+		}
+	case "all", "python", "java", "dotnet":
+		return fmt.Errorf("support for %q coming soon", lang)
+	default:
+		return fmt.Errorf("unsupported language %q", lang)
+	}
+
+	return nil
+}
+
+func publishToNPM(pkg, path string) error {
+	// verify path
+	_, err := os.Stat(path)
```

Comment:
What about other errors, are we ignoring them? If we are, can you add a comment explaining why?